Typescript error: Missing nonceManager in `mnemonicToAccount` options type

### Check existing issues

- [X] I checked there isn't [already an issue](https://github.com/wevm/viem/issues) for the bug I encountered.

### Viem Version

2.21.8

### Current Behavior

When providing the `nonceManager` option, Typescript gives an error even though it's a valid option:
```ts
import { mnemonicToAccount, nonceManager } from 'viem/accounts';

const account = mnemonicToAccount('some mnemonic...', { nonceManager });
```
```
error TS2353: Object literal may only specify known properties, and 'nonceManager' does not exist in type 'HDOptions'.
```

### Expected Behavior

Typescript should not give an error as `hdKeyToAccount` supports passing in a `nonceManager`:
```ts
export function mnemonicToAccount(
  mnemonic: string,
  opts: HDOptions = {},
): HDAccount {
  const seed = mnemonicToSeedSync(mnemonic)
  return hdKeyToAccount(HDKey.fromMasterSeed(seed), opts)
}
```

### Steps To Reproduce

_No response_

### Link to Minimal Reproducible Example

_No response_

### Anything else?

_No response_

--- a/.changeset/shaggy-ducks-give.md
+++ b/.changeset/shaggy-ducks-give.md
@@ -1,0 +1,5 @@
+---
+"viem": patch
+---
+
+Swapped defunct Ankr RPC URLs for Thirdweb

--- a/site/pages/docs/ethers-migration.mdx
+++ b/site/pages/docs/ethers-migration.mdx
@@ -47,7 +47,7 @@ Custom Chain:
 ```ts 
 import { providers } from 'ethers'
 
-const provider = new providers.JsonRpcProvider('https://rpc.ankr.com/fantom/​', { // [!code hl]
+const provider = new providers.JsonRpcProvider('https://250.rpc.thirdweb.com', { // [!code hl]
   name: 'Fantom', // [!code hl]
   id: 250 // [!code hl]
 }) // [!code hl]
@@ -74,7 +74,7 @@ import { fantom } from 'viem/chains'
 
 const client = createPublicClient({ // [!code hl]
   chain: fantom, // [!code hl]
-  transport: http('https://rpc.ankr.com/fantom/​') // [!code hl]
+  transport: http('https://250.rpc.thirdweb.com') // [!code hl]
 }) // [!code hl]
 ```
 

--- a/site/pages/docs/getting-started.mdx
+++ b/site/pages/docs/getting-started.mdx
@@ -46,7 +46,7 @@ const client = createPublicClient({ // [!code focus]
 ```
 
 :::info
-In a production app, it is highly recommended to pass through your authenticated RPC provider URL (Alchemy, Infura, Ankr, etc). If no URL is provided, viem will default to a public RPC provider. [Read more](/docs/clients/transports/http#usage).
+In a production app, it is highly recommended to pass through your authenticated RPC provider URL (Alchemy, Infura, Thirdweb, etc). If no URL is provided, viem will default to a public RPC provider. [Read more](/docs/clients/transports/http#usage).
 :::
 
 ### 2. Consume Actions

--- a/site/pages/experimental.mdx
+++ b/site/pages/experimental.mdx
@@ -31,7 +31,7 @@ const client = createWalletClient({ // [!code focus]
 ```
 
 :::info
-In a production app, it is highly recommended to pass through your authenticated RPC provider URL (Alchemy, Infura, Ankr, etc). If no URL is provided, viem will default to a public RPC provider. [Read more](/docs/clients/transports/http#usage).
+In a production app, it is highly recommended to pass through your authenticated RPC provider URL (Alchemy, Infura, Thirdweb, etc). If no URL is provided, viem will default to a public RPC provider. [Read more](/docs/clients/transports/http#usage).
 :::
 
 ### 2. Extend Client with Experimental Actions

--- a/site/pages/op-stack.mdx
+++ b/site/pages/op-stack.mdx
@@ -25,7 +25,7 @@ const client = createPublicClient({ // [!code focus]
 ```
 
 :::info
-In a production app, it is highly recommended to pass through your authenticated RPC provider URL (Alchemy, Infura, Ankr, etc). If no URL is provided, viem will default to a public RPC provider. [Read more](/docs/clients/transports/http#usage).
+In a production app, it is highly recommended to pass through your authenticated RPC provider URL (Alchemy, Infura, Thirdweb, etc). If no URL is provided, viem will default to a public RPC provider. [Read more](/docs/clients/transports/http#usage).
 :::
 
 ### 2. Extend Client with the OP Stack

--- a/src/chains/definitions/bsc.ts
+++ b/src/chains/definitions/bsc.ts
@@ -9,7 +9,7 @@ export const bsc = /*#__PURE__*/ defineChain({
     symbol: 'BNB',
   },
   rpcUrls: {
-    default: { http: ['https://rpc.ankr.com/bsc'] },
+    default: { http: ['https://56.rpc.thirdweb.com'] },
   },
   blockExplorers: {
     default: {

--- a/src/chains/definitions/btr.ts
+++ b/src/chains/definitions/btr.ts
@@ -10,11 +10,7 @@ export const btr = /*#__PURE__*/ defineChain({
   },
   rpcUrls: {
     default: {
-      http: [
-        'https://rpc.bitlayer.org',
-        'https://rpc.bitlayer-rpc.com',
-        'https://rpc.ankr.com/bitlayer',
-      ],
+      http: ['https://rpc.bitlayer.org', 'https://rpc.bitlayer-rpc.com'],
       webSocket: ['wss://ws.bitlayer.org', 'wss://ws.bitlayer-rpc.com'],
     },
   },

--- a/src/chains/definitions/chiliz.ts
+++ b/src/chains/definitions/chiliz.ts
@@ -11,10 +11,7 @@ export const chiliz = /*#__PURE__*/ defineChain({
   },
   rpcUrls: {
     default: {
-      http: [
-        'https://rpc.ankr.com/chiliz',
-        'https://chiliz-rpc.publicnode.com',
-      ],
+      http: ['https://chiliz-rpc.publicnode.com'],
     },
   },
   blockExplorers: {

--- a/src/chains/definitions/corn.ts
+++ b/src/chains/definitions/corn.ts
@@ -11,7 +11,7 @@ export const corn = /*#__PURE__*/ defineChain({
     symbol: 'BTCN',
   },
   rpcUrls: {
-    default: { http: ['https://rpc.ankr.com/corn_maizenet'] },
+    default: { http: ['https://21000000.rpc.thirdweb.com'] },
   },
   blockExplorers: {
     default: {

--- a/src/chains/definitions/cornTestnet.ts
+++ b/src/chains/definitions/cornTestnet.ts
@@ -11,7 +11,7 @@ export const cornTestnet = /*#__PURE__*/ defineChain({
     symbol: 'BTCN',
   },
   rpcUrls: {
-    default: { http: ['https://rpc.ankr.com/corn_testnet'] },
+    default: { http: ['https://21000001.rpc.thirdweb.com'] },
   },
   blockExplorers: {
     default: {

--- a/src/chains/definitions/fantom.ts
+++ b/src/chains/definitions/fantom.ts
@@ -9,7 +9,7 @@ export const fantom = /*#__PURE__*/ defineChain({
     symbol: 'FTM',
   },
   rpcUrls: {
-    default: { http: ['https://rpc.ankr.com/fantom'] },
+    default: { http: ['https://250.rpc.thirdweb.com'] },
   },
   blockExplorers: {
     default: {

--- a/src/chains/definitions/goerli.ts
+++ b/src/chains/definitions/goerli.ts
@@ -6,7 +6,7 @@ export const goerli = /*#__PURE__*/ defineChain({
   nativeCurrency: { name: 'Goerli Ether', symbol: 'ETH', decimals: 18 },
   rpcUrls: {
     default: {
-      http: ['https://rpc.ankr.com/eth_goerli'],
+      http: ['https://5.rpc.thirdweb.com'],
     },
   },
   blockExplorers: {

--- a/src/chains/definitions/harmonyOne.ts
+++ b/src/chains/definitions/harmonyOne.ts
@@ -9,7 +9,7 @@ export const harmonyOne = /*#__PURE__*/ defineChain({
     decimals: 18,
   },
   rpcUrls: {
-    default: { http: ['https://rpc.ankr.com/harmony'] },
+    default: { http: ['https://1666600000.rpc.thirdweb.com'] },
   },
   blockExplorers: {
     default: {

--- a/src/chains/definitions/polygonMumbai.ts
+++ b/src/chains/definitions/polygonMumbai.ts
@@ -6,7 +6,7 @@ export const polygonMumbai = /*#__PURE__*/ defineChain({
   nativeCurrency: { name: 'MATIC', symbol: 'MATIC', decimals: 18 },
   rpcUrls: {
     default: {
-      http: ['https://rpc.ankr.com/polygon_mumbai'],
+      http: ['https://80001.rpc.thirdweb.com'],
     },
   },
   blockExplorers: {

--- a/src/chains/definitions/swellchain.ts
+++ b/src/chains/definitions/swellchain.ts
@@ -8,10 +8,7 @@ export const swellchain = /*#__PURE__*/ defineChain({
   nativeCurrency: { name: 'Ether', symbol: 'ETH', decimals: 18 },
   rpcUrls: {
     default: {
-      http: [
-        'https://swell-mainnet.alt.technology',
-        'https://rpc.ankr.com/swell',
-      ],
+      http: ['https://swell-mainnet.alt.technology'],
     },
   },
   blockExplorers: {

--- a/src/utils/chain/defineChain.test.ts
+++ b/src/utils/chain/defineChain.test.ts
@@ -10,7 +10,7 @@ test('default', () => {
       network: 'celo',
       nativeCurrency: { name: 'Celo', symbol: 'CELO', decimals: 18 },
       rpcUrls: {
-        default: { http: ['https://rpc.ankr.com/celo'] },
+        default: { http: ['https://42220.rpc.thirdweb.com'] },
       },
     }),
   ).toMatchInlineSnapshot(`
@@ -28,7 +28,7 @@ test('default', () => {
         "rpcUrls": {
           "default": {
             "http": [
-              "https://rpc.ankr.com/celo",
+              "https://42220.rpc.thirdweb.com",
             ],
           },
         },

